lib/internal/url.js

```diff
@@ -227,7 +228,15 @@ class URLSearchParams {
           if (desc !== undefined && desc.enumerable) {
             const typedKey = toUSVString(key);
             const typedValue = toUSVString(init[key]);
-            this[searchParams].push(typedKey, typedValue);
+
+            // Two different key may result same after `toUSVString()`, we only
+            // leave the later one. Refers to WPT.
+            if (visited[typedKey] !== undefined) {
+              this[searchParams][visited[typedKey]] = typedValue;
+            } else {
+              this[searchParams].push(typedKey, typedValue);
+              visited[typedKey] = this[searchParams].length - 1;
```

Comment:
`<array>.push` returns the length already, maybe we can combine the two operations?
```suggestion
              visited[typedKey] = ArrayPrototypePush(this[searchParams], typedKey, typedValue) - 1;
```

Reply:
done.